design/repository-maintenance.md

```diff
@@ -0,0 +1,363 @@
+	cli.Get(context.Background(), client.ObjectKey{Namespace: f.Namespace(), Name: repo.Spec.BackupStorageLocation}, bsl)
+	if getError = checkError(err, logger); getError != nil {
+		return
+	}
+
+	para := provider.RepoParam{
+		BackupRepo:     repo,
+		BackupLocation: bsl,
+	}
+	// Connect
+	err = repoProvider.BoostRepoConnect(context.Background(), para)
+	if getError = checkError(err, logger); getError != nil {
+		return
+	}
+	// Prune repo
+	err = repoProvider.PruneRepo(context.Background(), para)
+	if getError = checkError(err, logger); getError != nil {
+		return
+	}
+}
+```
+
+### 2. Create Jobs by Repository Manager
+Currently, the backup repository controller will call the repository manager to do the `PruneRepo`, and Kopia or Restic maintenance is then finally called through multiple chain calls.
+
+We will keep using the `PruneRepo` function in the repository manager, but we cut off the multiple chain calls by creating a maintenance job.
+
+The job definition would be like below:
+```yaml
+apiVersion: v1
+items:
+- apiVersion: batch/v1
+  kind: Job
+  metadata:
+    creationTimestamp: "2024-02-19T02:22:36Z"
+    generation: 1
+    # labels or affinity or topology settings would inherit from the velero deployment
+    labels:
+      batch.kubernetes.io/controller-uid: fc7ab8a9-dcd4-4719-b4a2-f2fbcd091cd6
+      batch.kubernetes.io/job-name: nginx-example-default-kopia-pqz6c
+      controller-uid: fc7ab8a9-dcd4-4719-b4a2-f2fbcd091cd6
+      job-name: nginx-example-default-kopia-pqz6c
+    name: nginx-example-default-kopia-pqz6c
+    namespace: velero
+  spec:
+    backoffLimit: 1
+    completionMode: NonIndexed
+    completions: 1
+    parallelism: 1
+    selector:
+      matchLabels:
+        batch.kubernetes.io/controller-uid: fc7ab8a9-dcd4-4719-b4a2-f2fbcd091cd6
+    suspend: false
+    template:
+      metadata:
+        creationTimestamp: null
+        labels:
+          batch.kubernetes.io/controller-uid: fc7ab8a9-dcd4-4719-b4a2-f2fbcd091cd6
+          batch.kubernetes.io/job-name: nginx-example-default-kopia-pqz6c
+          controller-uid: fc7ab8a9-dcd4-4719-b4a2-f2fbcd091cd6
+          job-name: nginx-example-default-kopia-pqz6c
+        name: kopia-maintenance-job
+      spec:
+        containers:
+        # arguments for repo maintenance job
+        - args:
+          - server
+          - repo-maintenance
+          - --repo-name=nginx-example
+          - --repo-type=kopia
+          - --backup-storage-location=default
+          - --keep-latest-maintenance-jobs=3
+          - --log-level=debug
+          command:
+          - /velero
+          # inherit environment variables from the velero deployment
+          env:
+          - name: AZURE_CREDENTIALS_FILE
+            value: /credentials/cloud
+          # inherit image from the velero deployment
+          image: velero/velero:main
+          imagePullPolicy: IfNotPresent
+          name: kopia-maintenance-container
+          resources: {}
```

Comment:
If we deliberately choose not to set any resource limitation here, could you add a comment to explain?
i.e. if no resource limitation is set via velero server, velero will apply the "best effort" approach.

Reply:
Add some comments for resources field